src/in.c

```diff
@@ -344,6 +349,12 @@ ly_in_read(struct ly_in *in, void *buf, size_t count)
         return LY_EDENIED;
     }
 
+    for (size_t i = 0; i < count; i++) {
+        if (in->current[i] == '\n') {
+            LY_IN_NEW_LINE(in);
+        }
+    }
+
```

Comment:
The problem here was, that the lines were not counted correctly in many cases. Sometimes the parsers were skipping part of the input without considering that there can be a new line. I don't like this part either, I just wanted to fix it in the most simple way (do not mix it with other changes there).
But I agree - I'll add a separate commit to move the counting newlines back to the parsers but to all needed places - with the reference to `ly_in_read()` and `ly_in_skip()` it should be quite easy now. But it make sense that it should be part of the same PR.

Reply:
Okay. But for the use-case you mentioned (skipping characters without looking at them), I suppose function `ly_in_skip()` can count the newlines. Only `ly_in_read()` should not because it is expected the parser will look at the returned data. It may even make sense to add a parameter (flag) to both functions that will decide whether the newlines are counted or not. Something like `LY_IN_COUNT_NEWLINES` and `LY_IN_INGORE_NEWLINES`, for example, it would make it always obvious in the code. Just an idea, not sure if applicable.